test/jit/test_pdt.py

```diff
@@ -0,0 +1,189 @@
+        self.assertEqual(scripted_fn_bool((True, True, True)),
+                         test_list_and_tuple((True, True, True)))
+
+        scripted_fn_int = torch.jit._script_pdt(test_list_and_tuple, example_inputs=[((3, 4, 5), )])
+        self.assertEqual(scripted_fn_int((1, 2, 3)), test_list_and_tuple((1, 2, 3)))
+
+    def test_pdt_dict(self):
+        def test_dict(a):
+            return a['foo']
+
+        def test_dict_int_list(a):
+            return a[1]
+
+        make_global(test_dict, test_dict_int_list)
+
+        str_bool_inp = {'foo' : True, 'bar': False}
+        scripted_fn = torch.jit._script_pdt(test_dict, example_inputs=[(str_bool_inp,)])
+        self.assertEqual(scripted_fn({'foo' : False, 'bar': True}, ), test_dict({'foo' : False, 'bar': True}, ))
+
+        str_list_inp = {0 : [True, False], 1: [False, True]}
+        scripted_fn = torch.jit._script_pdt(test_dict_int_list, example_inputs=[(str_list_inp,)])
+        self.assertEqual(scripted_fn({0 : [False, False], 1: [True, True]}, ),
+                         test_dict_int_list({0 : [False, False], 1: [True, True]}, ))
+
+    def test_any(self):
+        def test_multiple_types(a):
+            assert not isinstance(a, bool)
+            return a
+
+        def test_multiple_type_refinement(a):
+            if isinstance(a, bool):
+                return 1
+            elif isinstance(a, int):
+                return 1 + a
+            elif isinstance(a, float):
+                return 1 + int(a)
+            else:
+                return -1
+
+        make_global(test_multiple_types, test_multiple_type_refinement)
+
+        scripted_fn = torch.jit._script_pdt(test_multiple_types, example_inputs=[(1,), ("abc", ), (8.9,), ([3, 4, 5], )])
+        self.assertEqual(scripted_fn(10), test_multiple_types(10))
+        self.assertEqual(scripted_fn("def"), test_multiple_types("def"))
+        self.assertEqual(scripted_fn(7.89999), test_multiple_types(7.89999))
+        self.assertEqual(scripted_fn([10, 11, 14]), test_multiple_types([10, 11, 14]))
+
+        scripted_fn = torch.jit._script_pdt(test_multiple_type_refinement, example_inputs=[(1,), ("abc", ), (8.9,),
+                                            ([3, 4, 5],), (True, ), ({"a": True}, ), ])
+        self.assertEqual(scripted_fn(10), test_multiple_type_refinement(10))
+        self.assertEqual(scripted_fn("def"), test_multiple_type_refinement("def"))
+        self.assertEqual(scripted_fn(7.89999), test_multiple_type_refinement(7.89999))
+        self.assertEqual(scripted_fn([10, 11, 14]), test_multiple_type_refinement([10, 11, 14]))
+        self.assertEqual(scripted_fn(False), test_multiple_type_refinement(False))
+        self.assertEqual(scripted_fn({"abc" : True, "def": False}), test_multiple_type_refinement({"abc" : True, "def": False}))
+
+    def test_class_as_profiled_types(self):
+        class UserDefinedClass:
+            def fn(self, b):
+                assert b is not None
```

Comment:
This statement doesn't really prove that `b`'s type is correctly inferred since all types can be compared against `None`.

Reply:
Fixed now.